src/App/Pages/Settings/ExportVaultPageViewModel.cs

```diff
@@ -0,0 +1,144 @@
+        }
+
+        public bool ShowPassword
+        {
+            get => _showPassword;
+            set => SetProperty(ref _showPassword, value,
+                additionalPropertyNames: new string[] {nameof(ShowPasswordIcon)});
+        }
+
+        public string MasterPassword
+        {
+            get => _masterPassword;
+            set => SetProperty(ref _masterPassword, value);
+        }
+
+        public Command TogglePasswordCommand { get; }
+
+        public string ShowPasswordIcon => ShowPassword ? "" : "";
+
+        public void TogglePassword()
+        {
+            ShowPassword = !ShowPassword;
+            (Page as ExportVaultPage).MasterPasswordEntry.Focus();
+        }
+
+        public Command ExportVaultCommand { get; }
+
+        public async Task ExportVaultAsync()
+        {
+            if(string.IsNullOrEmpty(_masterPassword))
+            {
+                _platformUtilsService.ShowToast("error", _i18nService.T("AnErrorHasOccurred"),
+                    _i18nService.T("InvalidMasterPassword"));
+                return;
+            }
+
+            var keyHash = await _cryptoService.HashPasswordAsync(_masterPassword, null);
+            MasterPassword = string.Empty;
+            
+            var storedKeyHash = await _cryptoService.GetKeyHashAsync();
+            if(storedKeyHash != null && keyHash != null && storedKeyHash == keyHash)
+            {
+                try
+                {
+                    var data = _exportService.GetExport(FileFormatOptions[FileFormatSelectedIndex].Key);
+                    var fileFormat = FileFormatOptions[FileFormatSelectedIndex].Key;
+                    _defaultFilename = _exportService.GetFileName(null, fileFormat);
+                    _exportResult = Encoding.ASCII.GetBytes(data.Result);
+
+                    if(!_deviceActionService.SaveFile(_exportResult, null, _defaultFilename, null))
+                    {
+                        ClearResult();
+                        await _platformUtilsService.ShowDialogAsync(_i18nService.T("ExportVaultFailure"));
+                    }
+                }
+                catch(Exception ex)
+                {
+                    ClearResult();
+                    System.Diagnostics.Debug.WriteLine(">>> {0}: {1}", ex.GetType(), ex.StackTrace);
+                }
+            }
+            else
+            {
+                _platformUtilsService.ShowToast("error", _i18nService.T("AnErrorHasOccurred"),
+                    _i18nService.T("InvalidMasterPassword"));
+            }
+        }
+
+        public async void SaveFileSelected(string contentUri, string filename)
+        {
+            if(_deviceActionService.SaveFile(_exportResult, null, filename ?? _defaultFilename, contentUri))
+            {
+                ClearResult();
+                _platformUtilsService.ShowToast("success", null, _i18nService.T("ExportVaultSuccess"));
+                return;
+            }
+
+            ClearResult();
```

Comment:
Looks like you call `ClearResult();` for both conditions, so just add it before the above `if`.

Reply:
The values being cleared are required in the `if` statement

Reply:
But it's called for both conditions, right?

Reply:
Correct

Reply:
So couldnt you do?

```
            
            ClearResult();
            if(_deviceActionService.SaveFile(_exportResult, null, filename ?? _defaultFilename, contentUri))
            {
                _platformUtilsService.ShowToast("success", null, _i18nService.T("ExportVaultSuccess"));
                return;
            }
            await _platformUtilsService.ShowDialogAsync(_i18nService.T("ExportVaultFailure"));
```

Reply:
Ignoring per clarification in chat